torch/fx/experimental/symbolic_shapes.py

```diff
@@ -591,11 +598,32 @@ def unary_magic_impl(self):
             return r.node
         # TODO: consider constant prop here
         expr = self.shape_env.replace(self.expr)
-        try:
-            out = func(expr)
-        except Exception:
-            log.warning(f"failed to eval {method}({expr})")
-            raise
+
+        # Attempt some extra simplification on SymInt
+        if method == "sym_int":
+            out = None
+            if isinstance(expr, sympy.Mul):
+                aa = expr.args
+                if len(aa) == 2 and isinstance(aa[0], sympy.Float) and aa[1].is_integer:
+                    coef = sympy.Integer(aa[0])
+                    if aa[0] == coef:  # structural equality test
```

Comment:
Hmm, I was wondering why you needed this, but I guess this is to test the scaling factor is not 2.5 and it's a nice integer. Maybe a comment would be nice.